From 3062f297078fb9a51001ba83971277a8bd1497ef Mon Sep 17 00:00:00 2001
From: Bo Yang <teboring@google.com>
Date: Wed, 13 Mar 2019 05:29:04 +0000
Subject: [PATCH 1/3] Fix macros in header of generated code

* Define PROTOBUF_MIN_PROTOC_VERSION in port_def.inc
* Remove duplicated GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION from common.h
---
 src/google/protobuf/port_def.inc   | 1 +
 src/google/protobuf/stubs/common.h | 4 ----
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/google/protobuf/port_def.inc b/src/google/protobuf/port_def.inc
index 9a6fc8c83f..278dd5c3d7 100644
--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -257,6 +257,7 @@
 
 #define PROTOBUF_VERSION 3007000
 #define PROTOBUF_MIN_HEADER_VERSION_FOR_PROTOC 3007000
+#define PROTOBUF_MIN_PROTOC_VERSION 3007000
 #define PROTOBUF_VERSION_SUFFIX ""
 
 // The minimum library version which works with the current version of the
diff --git a/src/google/protobuf/stubs/common.h b/src/google/protobuf/stubs/common.h
index 43da789f26..0b469c69c6 100644
--- a/src/google/protobuf/stubs/common.h
+++ b/src/google/protobuf/stubs/common.h
@@ -86,10 +86,6 @@ namespace internal {
 // A suffix string for alpha, beta or rc releases. Empty for stable releases.
 #define GOOGLE_PROTOBUF_VERSION_SUFFIX ""
 
-// The minimum library version which works with the current version of the
-// headers.
-#define GOOGLE_PROTOBUF_MIN_LIBRARY_VERSION 3007000
-
 // The minimum header version which works with the current version of
 // the library.  This constant should only be used by protoc's C++ code
 // generator.

From 8f9c46b5f9636cc30134f8d6cb27f26de3bf5829 Mon Sep 17 00:00:00 2001
From: Bo Yang <teboring@google.com>
Date: Thu, 14 Mar 2019 05:23:22 +0000
Subject: [PATCH 2/3] Send error if the error has been defined

---
 src/google/protobuf/port_def.inc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/google/protobuf/port_def.inc b/src/google/protobuf/port_def.inc
index 278dd5c3d7..c75e7a347a 100644
--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -97,6 +97,9 @@
 #ifdef PROTOBUF_MIN_HEADER_VERSION_FOR_PROTOC
 #error PROTOBUF_MIN_HEADER_VERSION_FOR_PROTOC was previously defined
 #endif
+#ifdef PROTOBUF_MIN_PROTOC_VERSION
+#error PROTOBUF_MIN_PROTOC_VERSION was previously defined
+#endif
 #ifdef PROTOBUF_PREDICT_TRUE
 #error PROTOBUF_PREDICT_TRUE was previously defined
 #endif

From 1426045c15daa29a3a03afa03e4e128787f11c51 Mon Sep 17 00:00:00 2001
From: Bo Yang <teboring@google.com>
Date: Fri, 15 Mar 2019 04:50:18 +0000
Subject: [PATCH 3/3] Undef the new macro

---
 src/google/protobuf/port_undef.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/google/protobuf/port_undef.inc b/src/google/protobuf/port_undef.inc
index b3177cdf31..6106c7dd4d 100644
--- a/src/google/protobuf/port_undef.inc
+++ b/src/google/protobuf/port_undef.inc
@@ -50,6 +50,7 @@
 #undef PROTOBUF_VERSION_SUFFIX
 #undef PROTOBUF_FIELD_OFFSET
 #undef PROTOBUF_MIN_HEADER_VERSION_FOR_PROTOC
+#undef PROTOBUF_MIN_PROTOC_VERSION
 #undef PROTOBUF_PREDICT_TRUE
 #undef PROTOBUF_PREDICT_FALSE
 #undef PROTOBUF_LONGLONG
